src/stdlib_stats_distribution_exponential.fypp

```diff
@@ -0,0 +1,322 @@
+        q = ve * exp(de)
+        ke(0) = int((de / q) * M2, kind = int32)
+        ke(1) = 0
+        we(0) = q / M2
+        we(255) = de / M2
+        fe(0) = ONE
+        fe(255) = exp(- de)
+        do  i = 254, 1, -1
+            de = -log(ve / de + exp(- de))
+            ke(i+1) = int(M2 * (de / te), kind = int32)
+            te = de
+            fe(i) = exp(- de)
+            we(i) = de / M2
+        end do
+        zig_exp_initialized = .true.
+    end subroutine zigset
+
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    function rvs_expon_0_${t1[0]}$${k1}$( ) result(res)
+    !
+    ! Standard exponential random variate (lambda=1)
+    !
+        ${t1}$ :: res, x
+        ${t1}$, parameter :: r = 7.69711747013104972_${k1}$
+        integer :: jz, iz
+
+        if(.not. zig_exp_initialized ) call zigset
+        iz = 0
+        jz = dist_rand(1_int32)                !32bit random integer
+        iz = iand( jz, 255 )                   !random integer in [0, 255]
+        if( abs( jz ) < ke(iz) ) then
+            res = abs(jz) * we(iz)
+        else
+            L1: do
+                if( iz == 0 ) then
+                    res = r - log( uni(1.0_${k1}$) )
+                    exit L1
+                end if
+                x = abs( jz ) * we(iz)
+                if(fe(iz) + uni(1.0_${k1}$) * (fe(iz-1) - fe(iz)) < exp(-x)) then
+                    res = x
+                    exit L1
+                end if
+                jz = dist_rand(1_int32)
+                iz = iand( jz, 255 )
+                if( abs( jz ) < ke(iz) ) then
+                    res = abs( jz ) * we(iz)
+                    exit L1
+                end if
+           end do L1
+       endif
+    end function rvs_expon_0_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    function rvs_expon_${t1[0]}$${k1}$(lambda) result(res)
+    !
+    ! Exponential distributed random variate
+    !
+        ${t1}$, intent(in) :: lambda
+        ${t1}$ :: res
+
+
+        if(lambda <= 0.0_${k1}$) call error_stop("Error(rvs_expon): Exponen"   &
+            //"tial distribution lambda parameter must be greater than zero")
+        res = rvs_expon_0_${t1[0]}$${k1}$(  )
+        res = res / lambda
+    end function rvs_expon_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    function rvs_expon_${t1[0]}$${k1}$(lambda) result(res)
+        ${t1}$, intent(in) :: lambda
+        ${t1}$ :: res
+        real(${k1}$) :: tr, ti
+
+        tr = rvs_expon_r${k1}$(lambda % re)
+        ti = rvs_expon_r${k1}$(lambda % im)
+        res = cmplx(tr, ti, kind=${k1}$)
+    end function rvs_expon_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    function rvs_expon_array_${t1[0]}$${k1}$(lambda, array_size) result(res)
+        ${t1}$, intent(in) :: lambda
+        integer, intent(in) :: array_size
+        ${t1}$ :: res(array_size), x, re
+        ${t1}$, parameter :: r = 7.69711747013104972_${k1}$
```

Comment:
What is this value? How is it computed?

Reply:
This parameter value was taken directly from the referenced paper, same as in the normal distribution.